llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -284,6 +284,7 @@ static bool mayBeVisibleThroughUnwinding(Value *V, Instruction *Start,
 
 void MemCpyOptPass::eraseInstruction(Instruction *I) {
   MSSAU->removeMemoryAccess(I);
+  EEI->removeInstruction(I);
```

Comment:
Should we call `MemCpyOptPass::eraseInstruction` in https://github.com/llvm/llvm-project/blob/a94060ca0c87d13af5c04410f3ed24c01d1de909/llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp#L1157? `NewCopySource` may be deleted after a BAA query.

Reply:
Nice catch, fixed!